### PR TITLE
Add new error type for transaction version mismatch

### PR DESCRIPTION
**Pull request by lovesh**
Signed-off-by: lovesh <lovesh.bond@gmail.com>

### CODE REVIEW

File: app/Makefile.version [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# This is the `transaction_version` field of `Runtime`
 APPVERSION_M=5
+# This is the `spec_version` field of `Runtime`
 APPVERSION_N=26
+# This is the `impl_version` field of `Runtime`
```
Comment (jleni) [inline app/Makefile.version:5]: This is not correct, we use this for internal patches.

File: app/src/parser_impl.c [hunk 2]
```diff
@@ -135,7 +137,7 @@ parser_error_t _readCompactInt(parser_context_t *c, compactInt_t *v) {
             CTX_CHECK_AND_ADVANCE(c, v->len)
             _getValue(v, &tmp);
             break;
-        case 3:         // bitint
+        case 3:         // bigint
```
Comment (jleni) [inline app/src/parser_impl.c:140]: thanks for fixing these typos!

**Review by jleni — CHANGES_REQUESTED** [review]
Thanks for fixing these typos.
Happy to merge, however, I would prefer if you can remove from the PR the submodule change. 
Modifying the SDK version has a much much bigger impact that the minor changes in this PR.

File: app/Makefile.version (hunk 1 above)
Comment (lovesh) [reply app/Makefile.version]: Thanks. Fixed.

**Comment by lovesh** [comment]
@jleni The changes to the submodule were unintentional, fixed them.

**Review by jleni — APPROVED** [review]

**Pull request merged**